### PR TITLE
feat: added check summary for component func

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves: https://github.com/flanksource/canary-checker/issues/1144

### CODE REVIEW

File: views/013_check_summary.sql [hunk 1]
```diff
@@ -44,3 +44,63 @@ CREATE
 OR REPLACE TRIGGER checks_last_transition_time BEFORE
 UPDATE ON checks FOR EACH ROW WHEN (OLD.status IS DISTINCT FROM NEW.status)
 EXECUTE PROCEDURE update_last_transition_time_for_check ();
+
+-- check summary view
+DROP VIEW IF EXISTS check_summary;
+
+CREATE VIEW check_summary AS
```
Comment (adityathebe) [inline views/013_check_summary.sql:51]: @moshloop the canary checker's `/api/summary` endpoint can simply call this.
Added it here instead so it's also usable in `check_summary_for_component`

Comment (moshloop) [reply views/013_check_summary.sql]: @adityathebe  we cannot include components in the default check_summary, You could have some checks linked to hundreds or thousands of components - especially when considering we might want to walk up or down the component hierarchy - i.e. Select a component and then show all health checks for related components.

Comment (adityathebe) [reply views/013_check_summary.sql]: @moshloop This is the same query that we've been using so far https://github.com/flanksource/duty/blob/main/check_summary.go#L16-L56. 
I just created a view out of it for reusability purpose.

Let me confirm if the `components` column is currently being used by the frontend.

Comment (moshloop) [reply views/013_check_summary.sql]: Can we also add:

*  check_summary_for_component_down : functions to return all checks below a component i.e. `path like component.path.%` 
*  check_summary_for_component_up : functions to return all checks above a component i.e. `where component_id in split(path, '.')`
*  check_summary_for_component_all: combination of up and down

**Pull request merged**